PiBox.Plugins/Jobs/Hangfire/src/PiBox.Plugins.Jobs.Hangfire/HangFirePlugin.cs

```diff
@@ -13,49 +15,57 @@
 using PiBox.Hosting.Abstractions.Services;
 using PiBox.Plugins.Jobs.Hangfire.Attributes;
 using PiBox.Plugins.Jobs.Hangfire.Job;
+using BindingFlags = System.Reflection.BindingFlags;
 
 namespace PiBox.Plugins.Jobs.Hangfire
 {
     public class HangFirePlugin : IPluginServiceConfiguration, IPluginApplicationConfiguration,
         IPluginHealthChecksConfiguration
     {
         private readonly IImplementationResolver _implementationResolver;
+        private readonly IHangfireConfigurator[] _configurators;
         private readonly HangfireConfiguration _hangfireConfig;
 
-        public HangFirePlugin(HangfireConfiguration configuration, IImplementationResolver implementationResolver)
+        public HangFirePlugin(HangfireConfiguration configuration, IImplementationResolver implementationResolver, IHangfireConfigurator[] configurators)
         {
             _implementationResolver = implementationResolver;
+            _configurators = configurators;
             _hangfireConfig = configuration;
         }
 
         public void ConfigureServices(IServiceCollection serviceCollection)
         {
             serviceCollection.AddFeatureManagement();
-            serviceCollection.AddSingleton<JobDetailCollection>();
-            serviceCollection.AddSingleton<IJobRegister>(sp => sp.GetRequiredService<JobDetailCollection>());
             serviceCollection.AddHangfire(conf =>
                 {
                     conf.UseSerializerSettings(new JsonSerializerSettings());
                     if (_hangfireConfig.InMemory)
-                    {
                         conf.UseMemoryStorage();
-                    }
                     else
-                    {
                         conf.UsePostgreSqlStorage(opts => opts.UseNpgsqlConnection(_hangfireConfig.ConnectionString));
```

Comment:
maybe we should make this configurable for other DB types someday

Reply:
just added the configurator so you would be able to overwrite it from the outside of the plugin